RedHatInsights/compliance-frontend
# RHICOMPL-611 - Refreshing on details view should not redirect

Before this change, the app was listening to the APP_NAVIGATION event in
redux for all pages. This meant that if you refresh on any details page,
the APP_NAVIGATION event will retrigger and try to redirect to the base
URL for that details page (e.g: /scappolicies for /scappolicies/:policy_id)

Instead, we should only listen to the events we care about, which are
the ones for moving to the vertical navigation items. All other links
through the app, shouldn't have to go through this.

Initially I was going to dismiss the APP_NAVIGATION events completely, as I've seen other apps do, but we can't because of the vertical navigation links, which rely on that. I took inspiration from the [advisor's App.js](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/App.js) to only react to these vertical-navigation link events. 

To test: 
* Ensure moving around the app still works, any links you can think of.
* Refresh on usual pages that didn't work before, confirm that it doesn't take you to anywhere else. Usual suspects include scap policy details, report details, reports by system, system details
